### ISSUE TITLE
HoloViews Holomap widgets not rendered when in a layout

### ISSUE THREAD

**Issue by maximlt** [holoviz/panel#5628]
When a HoloViews HoloMap is wrapped in a HoloViews pane that is passed to a layout, the widgets are not rendered.

```python
import holoviews as hv
import panel as pn

from bokeh.sampledata.penguins import data as df

hv.extension('bokeh')

ds = hv.Dataset(df, kdims=['bill_depth_mm', 'bill_length_mm', 'species'])
plot = ds.to(hv.Points, kdims=['bill_depth_mm', 'bill_length_mm'])
# pane = pn.pane.HoloViews(plot, sizing_mode='stretch_width')
pane = pn.pane.HoloViews(plot)

pane.servable()                            # OK: Widgets displayed
pn.Row('Test Row', pane).servable()        # Not OK: Widgets missing
pn.Column('Test Column', pane).servable()  # Not OK: Widgets missing
```

![image](https://github.com/holoviz/panel/assets/35924738/249bf675-1c7e-472f-b329-51322ed973bb)

**Comment by maximlt** [comment]
No errors in the console to be found.

**Comment by philippjfr** [comment]
I appreciate that this is pretty weird, but a HoloViews pane rendered on its own does not include the widgets (and never has). The simplest way to get the full layout is:

```python
pn.Row('Test Row', pane.layout).servable()
```

or alternatively:

```python
pn.Row('Test Row', pane, pane.widgets).servable()
```

Not opposed to finding a way to make this behave reasonably but suspect it might be difficult.

**Comment by maximlt** [comment]
Ok. I had no clue about that. I still don't understand what's the difference between my first example (`pane.servable()                            # OK: Widgets displayed`) and the other ones?

I thought it was a Panel/Bokeh/HoloViews bug. Apparently it's not so I'll have to make changes to hvPlot since now a Dynamic map doesn't render with its widgets in the Explorer (I guess it's because now the explorer preview is now in a Tabs layout).

**Comment by philippjfr** [comment]
>I still don't understand what's the difference between my first example (pane.servable() # OK: Widgets displayed) and the other ones?

Yeah, definitely quirky and hard to reason about. It's internal abstractions leaking. Internally Panel will unpack layouts if they are placed inside another layout but if they are rendered as a root then there's no layout that it could update and it ends up rendering the entire `.layout`. For most other components the `.layout` is simply a Column containing the one item that is being updated but for `HoloViews` it's the layout of the plot and the widgets.

**Comment by MarcSkovMadsen** [comment]
I believe I'm hit by the same. I was trying to create a custom data catalogue. Inspired by Intake I would like users to be able to dynamically change between customized plots for a given dataset. But getting it working in a proper way with and without `widgets`/ `groupby` seems not possible.

```python
import hvplot
import panel as pn
import hvplot.pandas
from bokeh.sampledata.penguins import data as df

hvplot.extension('plotly')

plot = pn.widgets.Select(value="default", options=["default", "other"], name="Plot")

def get_plot(name):
    if name=="other":
        return df.hvplot.scatter(x='bill_length_mm', y='bill_depth_mm', by="species", responsive=True)
    return df.hvplot.scatter(x='bill_length_mm', y='bill_depth_mm', groupby="species", responsive=True)

pn.Column(
    plot,
    pn.bind(get_plot, plot),
    pn.pane.HoloViews(pn.bind(get_plot, plot)),
    pn.pane.HoloViews(pn.bind(get_plot, plot)).layout,
).servable()
```

Below I've maked if it works as I would expect.

### Open the app

![image](https://github.com/holoviz/panel/assets/42288570/7a2064a1-15c5-4631-a76c-ffd8628a99bf)


### Change to `other` plot

![image](https://github.com/holoviz/panel/assets/42288570/a5777eb3-941a-45e6-aec4-e3c74d1ec818)

### Change back to `default` plot

![image](https://github.com/holoviz/panel/assets/42288570/4d3dace3-f888-4a6c-a20d-ef419e2bb08e)

**Comment by MarcSkovMadsen** [comment]
> I appreciate that this is pretty weird, but a HoloViews pane rendered on its own does not include the widgets (and never has). 

I believe I see a counter example in the `HoloViews` reference guide @philippjfr ?

![image](https://github.com/holoviz/panel/assets/42288570/6c072dd7-d6d7-4714-a622-44d6357b036e)

**Comment by philippjfr** [comment]
>I believe I see a counter example in the HoloViews reference guide @philippjfr ?

This is the bit I was talking about above:

> It's internal abstractions leaking. Internally Panel will unpack layouts if they are placed inside another layout but if they are rendered as a root then there's no layout that it could update and it ends up rendering the entire .layout.
